src/imagery/i.sam2/i.sam2.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python3
+
+############################################################################
+#
+# MODULE:       i.sam2
+# AUTHOR:       Corey T. White, OpenPlains Inc.
+# PURPOSE:      Uses the SAMGeo model for segmentation in GRASS GIS.
+# COPYRIGHT:    (C) 2023-2024 Corey White
+#               This program is free software under the GNU General
+#               Public License (>=v2). Read the file COPYING that
+#               comes with GRASS for details.
+#
+#############################################################################
+
+# %module
+# % description: Integrates SAMGeo model with text prompt for segmentation in GRASS GIS.
+# % keyword: imagery
+# % keyword: segmentation
+# % keyword: object recognition
+# % keyword: deep learning
+# %end
+
+# %option G_OPT_I_GROUP
+# % key: group
+# % description: Name of input imagery group
+# % required: yes
+# %end
+
+# %option
+# % key: output
+# % type: string
+# % description: Name of output segmented raster map
+# % gisprompt: new,cell,raster
+# % required: yes
+# %end
+
+# %option
+# % key: model_path
+# % type: string
+# % description: Path to the SAMGeo model file (optional if using default model)
+# % required: no
+# %end
+
+# %option
+# % key: text_prompt
+# % type: string
+# % description: Optional text prompt to guide segmentation
```

Comment:
The text prompt is not optional (required is set to 'yes')

Reply:
Yes. As of now the text prompt is not optional, so I have it set to required. We could however add additional functionality in the future that would make it optional.